python/ray/autoscaler/_private/providers.py

```diff
@@ -192,7 +192,24 @@ def _get_default_config(provider_config):
     package outside the autoscaler.
     """
     if provider_config["type"] == "external":
-        return {}
+        # Minimal config for compatibility with legacy-style external configs.
+        return {
+            "available_node_types": {
+                "head.empty": {
+                    "min_workers": 0,
+                    "max_workers": 0,
+                    "resources": {},
+                    "node_config": {},
+                },
+                "worker.empty": {
+                    "resources": {},
+                    "node_config": {},
+                }
+            },
+            "head_node_type": "head.empty",
+            "head_node": {},
+            "worker_nodes": {},
+        }
```

Comment:
why do we need this if prepare_config already does that?

Reply:
empty node_config also looks suspicious.

Reply:
The current logic of prepare_config assumes that a default yaml with this format has been provided.
This supplies the necessary default for legacy-style external configs.

Reply:
You're right about empty node_config -- I changed worker.empty and head.empty to empty dicts.
If you have a legacy-style external config with missing head or worker fields, it will now fail to validate the config.
Added test `testLegacyExternalNodeScalerMissingFields` to `test_autoscaler.py` to confirm that behavior.